[#17] 현재 진행 프로젝트 등록, 수정, 삭제

sileeee opened the pull request:
- 추가된 기능
    - 현재 진행 프로젝트 등록 / 수정 / 삭제

sileeee committed:
feat: 현재 진행 프로젝트 조회 기능 추가

sileeee committed:
feat: 현재 진행 프로젝트 등록, 수정, 삭제 기능 추가

Kyewon-Park commented on src/main/java/com/hri/hri_web_backend/controller/dto/RegisterProjectDto.java:
```diff
@@ -0,0 +1,21 @@
+public class RegisterProjectDto {
+
+	private String topic;
+
+	private String detail;
+
```
EX)
@Column(length = 500)
private String detail
같이 길이 지정해 주면 좋을 것 같아요. default값이 255라 넘을 일이 별로 없겠지만 혹시 넘으면 에러 날 수도 있어서요.

sileeee replied:
좋은 생각입니다~ 500자 정도면 프로젝트 설명으로도 충분할 것 같네요ㅎㅎ 수정해서 올리도록 하겠습니다!

Kyewon-Park commented on src/main/java/com/hri/hri_web_backend/service/ProjectService.java:
```diff
@@ -0,0 +1,49 @@
+		if(updateProject.isEmpty())
+			throw new NullPointerException();
+
+		updateProject.ifPresent(selectProject->{
+			selectProject.setTopic(dto.getTopic());
+			selectProject.setDetail(dto.getDetail());
+
+			projectRepository.save(selectProject);
+		});
```
projectRepository.save(selectProject); 는 없어도 될 것 같습니다!

sileeee replied:
감사합니다😊 수정하도록 하겠습니다!

Kyewon-Park commented on src/main/java/com/hri/hri_web_backend/domain/Project.java:
```diff
@@ -0,0 +1,36 @@
+
+	@Builder
+	public Project(Long id, String topic, String detail){
+		this.id = id;
```
this.id = id; 없어도 될 것 같습니다!

sileeee replied:
네 수정하겠습니다!

Kyewon-Park committed:
[#10] 학술지, 학술대회, 특허, 도서 소개 조회 ,등록 , 수정 및 삭제  (#18)

* feat: Publication 조회, 생성, 수정 및 삭제

Publication 조회, 생성, 수정 및 삭제
SuccessResponse, redirect 사용

* refactor: 엔티티 setter추가

* fix: ApplicationContext 오류 고침/ publicationService 테스트 완료

failed to load applicationcontext spring boot test
=> PublicationRepository 에서 _언더바 인식 안됨 => Entity의 publication_type 을 publicationType으로 고침

수정/삭제 테스트 추가 및 실행 완료

* style: publication service/repo 이름 및 테스트 클래스 이름 수정

+ else 문 삭제

* Create PublicationFixture.java

sileeee committed:
refactor: 폴더 구조 정리

sileeee committed:
refactor: detail 제한길이 지정

sileeee committed:
fix: id setter삭제

Kyewon-Park committed:
[#12][#14] 관리자 로그인 검증 로직 / redis session storage 연동  (#20)

* feat: Spring Security, JWT (로그인 검증) / Redis 추가

AuthService 추가 : Salt값 이용하여 사용자 패스워드 암호화 로그인,
JWT 통해서 Access token 과 Refresh token 발행(token은 Cookie 형태로 저장)

* feat: POSTMAN 테스팅 완료, WebSecurityConfig 파일 수정, 컨트롤러 경로 수정

* POSTMAN 테스팅 완료

* WebSecurityConfig 파일 수정 : http jwtRequestFilter 추가, 접근 가능 url 수정
- 로그인 이후 access/refresh토큰 받아 권한 인증 후 api사용

* 컨트롤러 경로 수정 : admin 권한이 필요한 api는 /admin 추가

* Update application.properties

* style: 필요없는 import문 제거

* style: init코드 변경, 기존 테스트 삭제

* fix: url명 수정

Kyewon-Park committed:
[#12][#14] 관리자 로그인 검증 로직 / redis session storage 연동  (#20)

* feat: Spring Security, JWT (로그인 검증) / Redis 추가

AuthService 추가 : Salt값 이용하여 사용자 패스워드 암호화 로그인,
JWT 통해서 Access token 과 Refresh token 발행(token은 Cookie 형태로 저장)

* feat: POSTMAN 테스팅 완료, WebSecurityConfig 파일 수정, 컨트롤러 경로 수정

* POSTMAN 테스팅 완료

* WebSecurityConfig 파일 수정 : http jwtRequestFilter 추가, 접근 가능 url 수정
- 로그인 이후 access/refresh토큰 받아 권한 인증 후 api사용

* 컨트롤러 경로 수정 : admin 권한이 필요한 api는 /admin 추가

* Update application.properties

* style: 필요없는 import문 제거

* style: init코드 변경, 기존 테스트 삭제

* fix: url명 수정

sileeee committed:
Merge branch 'develop' of https://github.com/HRI-WebProject/HRI-Homepage-Backend into feature/17

 Conflicts:
	src/main/java/com/hri/hri_web_backend/controller/AdminController.java
	src/main/java/com/hri/hri_web_backend/controller/MemberController.java
	src/main/java/com/hri/hri_web_backend/domain/Salt.java
	src/main/java/com/hri/hri_web_backend/service/MemberService.java

sileeee committed:
fix: conflict 해결

sileeee committed:
Merge branch 'feature/12' into feature/17

sileeee committed:
fix: 구성원 수정 테스트코드 삭제

sileeee committed:
Merge remote-tracking branch 'origin/feature/17' into feature/17

# Conflicts:
#	src/main/java/com/hri/hri_web_backend/controller/MemberController.java
#	src/test/java/com/hri/hri_web_backend/service/MemberServiceTest.java

sileeee committed:
feat: 프로젝트 CRUD 테스트코드 추가

Kyewon-Park committed:
feat: Member Update 테스트코드 추가
